pkg/util/log/exit_override.go

```diff
@@ -12,13 +12,40 @@ package log
 
 import (
 	"context"
+	"sync"
 
 	"github.com/cockroachdb/cockroach/pkg/cli/exit"
 	"github.com/cockroachdb/cockroach/pkg/util/log/channel"
 	"github.com/cockroachdb/cockroach/pkg/util/log/logpb"
 	"github.com/cockroachdb/cockroach/pkg/util/log/severity"
 )
 
+var emergencyStopFunc struct {
+	sync.Mutex
+	fn func()
+}
+
+// EmergencyStop invokes the emergency stop function set through
+// SetEmergencyStopFunc, if any. EmergencyStop is a hack to close network
+// connections in the event of a disk stall that may prevent the process from
+// exiting.
+func EmergencyStop() {
```

Comment:
I realize we shouldn't really "log", but I'm wondering if there's some way of leaving a paper trail that we got in here and did the emergency stop?